LoopKit/InsulinKit/DoseEntry.swift

```diff
@@ -18,7 +18,13 @@ public struct DoseEntry: TimelineValue, Equatable {
     public let unit: DoseUnit
     public let deliveredUnits: Double?
     public let description: String?
+    public let insulinModel: InsulinModel?
     internal(set) public var syncIdentifier: String?
+    
+    // ANNA TODO: this is needed due to the fact the insulin model doesn't conform to equatable; would it be possible to avoid having to specify this?
+    public static func == (lhs: DoseEntry, rhs: DoseEntry) -> Bool {
+        return lhs.type == rhs.type && lhs.startDate == rhs.startDate && lhs.endDate == rhs.endDate && lhs.value == rhs.value && lhs.deliveredUnits == rhs.deliveredUnits && lhs.description == rhs.description && lhs.insulinModel?.effectDuration == rhs.insulinModel?.effectDuration && lhs.insulinModel?.delay == rhs.insulinModel?.delay
+    }
```

Comment:
Is there a way to make InsulinModel conform to equatable so this override isn't needed?